frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -1034,35 +1038,177 @@ export const ClusterServiceVersionDetails: React.SFC<ClusterServiceVersionDetail
   );
 };
 
-export const CSVSubscription: React.FC<CSVSubscriptionProps> = ({
-  obj,
-  subscriptions = [],
-  ...rest
-}) => {
+export const CSVSubscription: React.FC<CSVSubscriptionProps> = ({ obj, customData }) => {
   const EmptyMsg = () => (
     <MsgBox title="No Operator Subscription" detail="This Operator will not receive updates." />
   );
 
-  const subscription = React.useMemo(() => subscriptionForCSV(subscriptions, obj), [
-    obj,
-    subscriptions,
-  ]);
+  const [
+    { subscription, catalogSource, installPlan, packageManifest },
+    loaded,
+    loadError,
+  ] = customData;
 
   return (
-    <StatusBox EmptyMsg={EmptyMsg} loaded data={subscription}>
+    <StatusBox EmptyMsg={EmptyMsg} loaded={loaded} loadError={loadError} data={subscription}>
       <SubscriptionDetails
-        {...rest}
+        catalogSources={[catalogSource]}
+        installPlans={[installPlan]}
+        packageManifests={[packageManifest]}
         obj={subscription}
+        subscriptions={[subscription]}
         clusterServiceVersions={[obj]}
-        subscriptions={subscriptions}
       />
     </StatusBox>
   );
 };
 
+type DataLoaderProps = {
+  name: string;
+  namespace: string;
+};
+
+type DataLoaderResults = [
+  {
+    packageManifest: PackageManifestKind;
+    installPlan: InstallPlanKind;
+    catalogSource: CatalogSourceKind;
+    subscription: SubscriptionKind;
+    csv: ClusterServiceVersionKind;
+  },
+  boolean,
+  any,
+];
+
+const useCSVPageDataLoader = (dataLoaderProps: DataLoaderProps): DataLoaderResults => {
+  const { name, namespace } = dataLoaderProps;
+  const csvResource = {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    name,
+    namespace,
+    isList: false,
+  };
+
+  const [csvData, csvLoaded, csvLoadError] = useK8sWatchResource<ClusterServiceVersionKind>(
+    csvResource,
+  );
+
+  const memoizedCSVData = useDeepCompareMemoize(csvData, true);
+
+  const operatorNamespace = memoizedCSVData?.metadata?.annotations?.['olm.operatorNamesapce'];
+
+  const subscriptionResource = {
+    isList: true,
+    kind: referenceForModel(SubscriptionModel),
+    namespace: operatorNamespace,
+  };
```

Comment:
We can use fieldSelector here to get only the subscription we need:

```suggestion
  const subscriptionResource = {
    isList: false,
    kind: referenceForModel(SubscriptionModel),
    namespace: operatorNamespace,
    fieldSelector: `status.installedCSV=${name}`
  };
```